CMakeLists.txt

```diff
@@ -158,6 +159,48 @@ include(cmake/leveldb.cmake)
 include(cmake/minisketch.cmake)
 include(cmake/secp256k1.cmake)
 
+include(ProcessConfigurations)
+set_default_config(RelWithDebInfo)
+
+# Redefine configuration flags.
+add_compile_definitions($<$<CONFIG:Debug>:DEBUG>)
+add_compile_definitions($<$<CONFIG:Debug>:DEBUG_LOCKORDER>)
+add_compile_definitions($<$<CONFIG:Debug>:DEBUG_LOCKCONTENTION>)
+add_compile_definitions($<$<CONFIG:Debug>:RPC_DOC_CHECK>)
+add_compile_definitions($<$<CONFIG:Debug>:ABORT_ON_FAILED_ASSUME>)
+# We leave assertions on.
+if(MSVC)
+  remove_c_flag_from_all_configs(/DNDEBUG)
+  remove_cxx_flag_from_all_configs(/DNDEBUG)
+else()
+  remove_c_flag_from_all_configs(-DNDEBUG)
+  remove_cxx_flag_from_all_configs(-DNDEBUG)
+
+  # Prefer -O2 optimization level. (-O3 is CMake's default for Release for many compilers.)
+  replace_c_flag_in_config(Release -O3 -O2)
+  replace_cxx_flag_in_config(Release -O3 -O2)
```

Comment:
I believe this means the user has no way to override and is stuck with "-O2" builds ?

Reply:
FWIW, an issue with the user's opportunities to override build system options described in details in https://github.com/bitcoin-core/secp256k1/pull/1249.